pkg/client/cache/delta_fifo.go

```diff
@@ -132,6 +135,12 @@ var (
 	ErrZeroLengthDeltasObject = errors.New("0 length Deltas object; can't get key")
 )
 
+// Close the queue.
+func (f *DeltaFIFO) Close() {
+	f.closed = true
```

Comment:
You are missing a lock here, otherwise there is no guarantee closed is published.